### ISSUE TITLE
Experienced a sudden drop in train loss

### ISSUE THREAD

**Issue by ftakanashi** [jadore801120/attention-is-all-you-need-pytorch#94]
A great project for me to learn about transformer.
But a very strange thing happened when I try to train the model.

As my GPU's RAM is rather small, I modified batch_size to 16 and retained all other hyper parameters. After training started, the first epoch went well and output an training accuracy (per word) about 0.45. But at some point of the second epoch (usually the end of the first quarter of the second epoch), training accuracy will suddenly drop to 0.2 or even lower.
Then, the longer train lasts, the lower accuracy becomes. I'm very curious about why that happens.

**Comment by wanwanaa** [comment]
I also have this problem, have you solved the problem yet?

**Comment by one-tree** [comment]
Maybe your batch_size is too small.I tried to change the batch_size to 256, and found that the accuracy was first increased, then decreased. But this turning point is more backward than when it's 128.
